### ISSUE TITLE
Nested !If in target group attributes gets false error E3012

### ISSUE THREAD

**Issue by pnickerson-cashstar** [aws-cloudformation/cfn-lint#4284]
### CloudFormation Lint Version

1.40.4

### What operating system are you using?

Fedora Linux 42

### Describe the bug

If I have nested `!If` intrinsic functions in an `AWS::ElasticLoadBalancingV2::TargetGroup` resource's `TargetGroupAttributes` property, then `cfn-lint` is falsely reporting error `E3012`:
```
E3012 [{'Key': 'stickiness.type', 'Value': 'lb_cookie'}, {'Key': 'stickiness.lb_cookie.duration_seconds', 'Value': 86400}] is not of type 'object'
testing-cfn-lint.yaml:46:15

E3012 [{'Key': 'stickiness.type', 'Value': 'source_ip'}] is not of type 'object'
testing-cfn-lint.yaml:50:15
```

### Expected behavior

I've checked the reproduction template with `aws cloudformation validate-template`, `rain format`, and `yamllint`. And I am able to use it to create a stack. As far as I can tell, it is a valid template. So I would expect `cfn-lint` to return no errors.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Parameters:
  ParameterA:
    Type: String
    AllowedValues:
      - ValueB
      - ValueC

  ParameterD:
    Type: String
    AllowedValues:
      - ValueE
      - ValueF

Conditions:
  ConditionG: !Equals
    - !Ref ParameterA
    - ValueB

  ConditionH: !Equals
    - !Ref ParameterD
    - ValueE

Resources:
  VPC:
    Type: AWS::EC2::VPC
    Properties:
      CidrBlock: 10.0.0.0/23

  TargetGroup:
    Type: AWS::ElasticLoadBalancingV2::TargetGroup
    Properties:
      Protocol: HTTPS
      Port: 443
      VpcId: !Ref VPC
      TargetGroupAttributes: !If
        - ConditionG
        - !Ref AWS::NoValue
        - - Key: deregistration_delay.timeout_seconds
            Value: 30
          - Key: stickiness.enabled
            Value: "false"
          - !If
            - ConditionH
            - - Key: stickiness.type
                Value: lb_cookie
              - Key: stickiness.lb_cookie.duration_seconds
                Value: 86400
            - - Key: stickiness.type
                Value: source_ip
```

**Comment by pnickerson-cashstar** [comment]
Not completed. Should be not planned.
